### ISSUE TITLE
Vagrant 2.2.0 does not support guest NethServer 7

### ISSUE THREAD

**Issue by syntaxerrormmm** [hashicorp/vagrant#10391]
### Vagrant version
```
$ vagrant -v
Vagrant 2.2.0
```

### Host operating system
```
$ cat /etc/lsb-release 
DISTRIB_ID=Zorin
DISTRIB_RELEASE=12
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Zorin OS 12.4"
$ uname -a 
Linux syntaxxps 4.15.0-38-generic #41~16.04.1-Ubuntu SMP Wed Oct 10 20:16:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ dpkg -l | grep libvirt
ii  gir1.2-libvirt-glib-1.0:amd64                        0.2.2-0.1ubuntu1                                                            amd64        libvirt glib mainloop integration
ii  libvirt-bin                                          1.3.1-1ubuntu10.24                                                          amd64        programs for the libvirt library
ii  libvirt-dev:amd64                                    1.3.1-1ubuntu10.24                                                          amd64        development files for the libvirt library
ii  libvirt-glib-1.0-0:amd64                             0.2.2-0.1ubuntu1                                                            amd64        libvirt glib mainloop integration
ii  libvirt0:amd64                                       1.3.1-1ubuntu10.24                                                          amd64        library for interfacing with different virtualization systems
ii  python-libvirt                                       1.3.1-1ubuntu1.1                                                            amd64        libvirt Python bindings
```

### Guest operating system
NethServer 7.5.1804 - A CentOS7-based distro; box is a custom libvirt-based one, built manually from iso images that can be downloaded from official site - https://www.nethserver.org/
```
$  cat /etc/redhat-release 
CentOS Linux release 7.5.1804 (Core) 
$ uname -a
Linux ns7-box.localdomain 3.10.0-862.11.6.el7.x86_64 #1 SMP Tue Aug 14 21:49:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$  systemctl status NetworkManager
● NetworkManager.service - Network Manager
   Loaded: loaded (/usr/lib/systemd/system/NetworkManager.service; disabled; vendor preset: disabled)
   Active: inactive (dead)
     Docs: man:NetworkManager(8)
```

### Vagrantfile
```
Vagrant.configure("2") do |config|
  config.vm.box = "syntaxerrormmm/nethserver-7"
  config.vm.box_url = "https://vagrant.sys42.eu/syntaxerrormmm/nethserver-7"
end
```

### Debug output
https://gist.github.com/a431a2ff4e7b7a6a2adb2ff47b0430f3

### Expected behavior
Should run a new box based on NethServer 7.

### Actual behavior
Box is working but `vagrant up` breaks and terminates with exit code 1 and the following error:
```
==> test_ns7plain_default: Configuring and enabling network interfaces...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

(test -f /etc/init.d/NetworkManager && /etc/init.d/NetworkManager restart) || ((systemctl | grep NetworkManager.service) && systemctl restart NetworkManager)

Stdout from the command:



Stderr from the command:


```

**Works fine with 2.1.5**.

### Steps to reproduce
1. Configure libvirt environment (I used virt-manager)
2. Configure a network, host-only, with base address 172.16.253.0/24
3. Install vagrant 2.2.0
4. Install vagrant-libvirt
5. Create Vagrantfile reported
6. vagrant up --provider=libvirt

Enjoy the mayhem.

### Notes

**Comment by tuminoid** [comment]
This is also an issue with regular RHEL or CentOS which you install from scratch and do not include NetworkManager. Vagrant 1.9.x works fine, but 2.x has removed the check if NetworkManager is installed.

**Comment by tuminoid** [comment]
TL;DR: If you add any extra network to Vagrant, like private network, it'll cause Vagrant to ignore the return value of NetworkManager related commands, and bringing up the box will succeed.

If you look at `plugins/guests/redhat/cap/configure_networks.rb` at line 69 and down, you can see Vagrant is adding up commands to run to `commands[:start]`, `commands[:middle]` and `commands[:end]`. NetworkManager related commands get appended to `commands[:middle]`. Then those will be concatenated in line 83, and executed.

If you don't have extra interface, there is no commands in `commands[:end]`, which means the NetworkManager command at `:middle` is the last one to run, and its return value is used, and bring up will fail due to return code 1. However, if you add any interfaces, they append commands to `commands[:end]`, and return value from those commands will be used. Any command that fails in `:start` or `:middle` will just be ignored.

This is equivalent to running bash script without `set -e`. Maybe @briancain can comment if this is really the intention.

**Comment by tuminoid** [comment]
> This is also an issue with regular RHEL or CentOS which you install from scratch and do not include NetworkManager. Vagrant 1.9.x works fine, but 2.x has removed the check if NetworkManager is installed.

Also, if NetworkManager is not installed, this also wouldn't occur, as Vagrant checks if `nmcli` is installed (which is part of NetworkManager package). In our case, we actually just disable NetworkManager, which creates the issue by system having `nmcli`, but `systemctl | grep NetworkManager.service` returns nothing as NM is disabled.

The actual code changes related to this were merged for v1.9.5.

**Comment by syntaxerrormmm** [comment]
> TL;DR: If you add any extra network to Vagrant, like private network, it'll cause Vagrant to ignore the return value of NetworkManager related commands, and bringing up the box will succeed.

I can confirm it on vagrant 2.2.5 and a new version of the NethServer image, 7.6.1811.

The case here is that NM is installed but disabled by default. I admit specifying a not-needed interface is a hackish workaround; I don't know if adding an option inside the Vagrantfile that would insert some commands in the `commands[:end]` or simply ignore exiting errors on NM is the way to go, but I think a `ignore_nm_errors :true` inside Vagrantfile could be a good starting point. Cannot provide any patch though, sorry.
